pkgs/development/cuda-modules/packages/removeStubsFromRunpathHook/removeStubsFromRunpathHook.bash

```diff
@@ -0,0 +1,101 @@
+# shellcheck shell=bash
+
+# Only run the hook from nativeBuildInputs when strictDeps is set
+if [[ -n ${removeStubsFromRunpathHookOnce-} ]]; then
+  nixDebugLog "skipping sourcing removeStubsFromRunpathHook.bash (hostOffset=${hostOffset:-0}) (targetOffset=${targetOffset:-0})" \
+    "because it has already been sourced"
+  return 0
+elif [[ -n ${strictDeps:-} ]] && ((${hostOffset:-0} != -1 || ${targetOffset:-0} != 0)); then
+  nixDebugLog "skipping sourcing removeStubsFromRunpathHook.bash (hostOffset=${hostOffset:-0}) (targetOffset=${targetOffset:-0})" \
+    "because it is not in depsBuildBuild"
+  return 0
+fi
+
+declare -g removeStubsFromRunpathHookOnce=1
+
+nixLog "sourcing removeStubsFromRunpathHook.bash (hostOffset=${hostOffset:-0}) (targetOffset=${targetOffset:-0})"
+
+# NOTE: Adding to prePhases to ensure all setup hooks are sourced prior to adding our hook.
+appendToVar prePhases removeStubsFromRunpathHookRegistration
+nixLog "added removeStubsFromRunpathHookRegistration to prePhases"
+
+# Registering during prePhases ensures that all setup hooks are sourced prior to installing ours,
+# allowing us to always go after autoAddDriverRunpath and autoPatchelfHook.
+removeStubsFromRunpathHookRegistration() {
+  local postFixupHook
+
+  # Check if "autoFixElfFiles addDriverRunpath" is in postFixupHooks.
+  # If it is, warn the user about it -- it does not play well with other hooks modifying the runpath and should be
+  # unnecessary when linking against stub files (which is the only reason this setup hook would be sourced), since
+  # the stub runpath entries are replaced with the driver link.
+  for postFixupHook in "${postFixupHooks[@]}"; do
+    if [[ $postFixupHook == "autoFixElfFiles addDriverRunpath" ]]; then
+      nixLog "discovered 'autoFixElfFiles addDriverRunpath' in postFixupHooks; this hook should be unnecessary when" \
+        "linking against stub files!"
+    fi
+  done
+
+  # NOTE: We assume postFixupHooks is an array to abuse calling convention, allowing us to call the higher-order
```

Comment:
It is declared and used as an array, I don't think there's any abuse here

Reply:
I was thinking of `phases` that doesn't allow spaces, hooks are fine: https://github.com/NixOS/nixpkgs/blob/4618896043f5a709735bade97033b22fc0b1019c/doc/stdenv/stdenv.chapter.md?plain=1#L581.

Will remove.